src/renderer/components/NewConnectionForm.tsx

```diff
@@ -0,0 +1,312 @@
+              <Form.Group className="mb-2" controlId="connectionAddress">
+                <Form.Label>Address</Form.Label>
+                <Form.Control
+                  className="form-control-sm"
+                  value={form.address}
+                  onChange={(e) => setField('address', e.target.value)}
+                  type="text"
+                  placeholder="Ex: localhost"
+                  isInvalid={!!errors.address}
+                />
+                <Form.Control.Feedback type="invalid">
+                  {errors.address}
+                </Form.Control.Feedback>
+              </Form.Group>
+              <Form.Group className="mb-2" controlId="connectionPort">
+                <Form.Label>Port</Form.Label>
+                <Form.Control
+                  className="form-control-sm"
+                  value={form.port}
+                  onChange={(e) => setField('port', e.target.value)}
+                  type="number"
+                  placeholder="Ex: 3000"
+                  isInvalid={!!errors.port}
+                />
+                <Form.Control.Feedback type="invalid">
+                  {errors.port}
+                </Form.Control.Feedback>
+              </Form.Group>
+              <Form.Group className="mb-2" controlId="connectionUser">
+                <Form.Label>Username</Form.Label>
+                <Form.Control
+                  className="form-control-sm"
+                  value={form.username}
+                  onChange={(e) => setField('username', e.target.value)}
+                  type="text"
+                  placeholder="Username"
+                  isInvalid={!!errors.username}
+                />
+                <Form.Control.Feedback type="invalid">
+                  {errors.username}
+                </Form.Control.Feedback>
+              </Form.Group>
+              <Form.Group className="mb-2" controlId="connectionPassword">
+                <Form.Label>Password</Form.Label>
+                <Form.Control
+                  className="form-control-sm"
+                  value={form.password}
+                  onChange={(e) => setField('password', e.target.value)}
+                  type="password"
+                  placeholder="Password"
+                  isInvalid={!!errors.password}
+                />
+                <Form.Control.Feedback type="invalid">
+                  {errors.password}
+                </Form.Control.Feedback>
+              </Form.Group>
+            </>
+          )}
+          {connectionStringActive && (
+            <Form.Group className="mb-2" controlId="connectionString">
+              <Form.Label>Connection String</Form.Label>
+              <Form.Control
+                className="form-control-sm"
+                value={form.connectionString}
+                onChange={(e) => setField('connectionString', e.target.value)}
+                type="text"
+                placeholder="Ex: postgresql://username:password@address:port"
```

Comment:
Highlighting the example text that I'm unsure about

Reply:
It's of the format `postgresql://dbuser:secretpassword@database.server.com:3211/mydb`